src/Service/StatisticsExtractionService.php

```diff
@@ -0,0 +1,271 @@
+ */
+class StatisticsExtractionService
+{
+    protected const BATCH_SIZE = 50;
+
+    private $documentManager;
+    private $logger;
+    private $extractionResultRepository;
+    private $elasticsearchService;
+    private $entryRepository;
+
+    /**
+     * StatisticsExtractionService constructor.
+     *
+     * @param \Doctrine\ODM\MongoDB\DocumentManager $documentManager
+     *   The document manager
+     * @param \App\Repository\EntryRepository $entryRepository
+     *   Repository for Entry documents
+     * @param \App\Repository\ExtractionResultRepository $extractionResultRepository
+     *   Repository for ExtractResult documents
+     * @param \Psr\Log\LoggerInterface $logger
+     *   The logger
+     * @param \App\Service\ElasticsearchServiceInterface $elasticsearchService
+     *   Service to integrate with elasticsearch
+     */
+    public function __construct(DocumentManager $documentManager, EntryRepository $entryRepository, ExtractionResultRepository $extractionResultRepository, LoggerInterface $logger, ElasticsearchServiceInterface $elasticsearchService)
+    {
+        $this->documentManager = $documentManager;
+        $this->logger = $logger;
+        $this->elasticsearchService = $elasticsearchService;
+        $this->extractionResultRepository = $extractionResultRepository;
+        $this->entryRepository = $entryRepository;
+    }
+
+    /**
+     * Extract new statistics.
+     *
+     * @throws \Doctrine\ODM\MongoDB\MongoDBException
+     * @throws \Throwable
+     */
+    public function extractStatistics()
+    {
+        $today = new \DateTime();
+
+        // Get latest extraction entry. Default to first day of production.
+        /* @var ExtractionResult $lastExtraction */
+        $lastExtraction = $this->extractionResultRepository->getNewestEntry();
+
+        // Default to 1. december 2019 to make sure we extract all statistics from the start of production.
+        /* @var \DateTime $latestExtractionDate */
+        $latestExtractionDate = $lastExtraction ? $lastExtraction->getDate() : new \Datetime('1 december 2019');
+
+        $numberOfDaysToSearch = (int) $today->diff($latestExtractionDate)->format('%a');
+
+        $numberOfEntriesAdded = 0;
+
+        // Extract logs for all dates from latest extraction date to yesterday.
+        // Create all as Entry documents in the database.
+        while ($numberOfDaysToSearch > 0) {
+            $dayToSearch = new \DateTime('-'.($numberOfDaysToSearch - 1).' days');
+
+            // Get entries from elasticsearch index.
+            $statistics = $this->elasticsearchService->getLogsFromElasticsearch($dayToSearch, 'Cover request/response');
+
+            $nextBatchLimit = self::BATCH_SIZE;
+
+            // Add all to mongodb.
+            foreach ($statistics as $statisticsEntry) {
+                // Flush when batch size is exceeded to avoid memory buildup.
+                if (0 < $numberOfEntriesAdded && $numberOfEntriesAdded > $nextBatchLimit) {
```

Comment:
Could this be simplified if it was place at the end og the loop (then `$numberOfEntriesAdded` would properly never be `0`)

Reply:
Fixed.
I have added it to the start of the loop, because a variable number of entries are added for each iteration of the loop. This way the `if`-block is only added one place, instead of before each `continue`.